### ISSUE TITLE
Keyboard Dismisses Unexpectedly During Order Search

### ISSUE THREAD

**Issue by mdrockwell** [woocommerce/woocommerce-android#3616]
**Describe the bug**
When using the search feature on the orders tab, the keyboard is dismissed mid-search to display results from the characters that have been entered into the search field. This can happen before you finish entering your search term.

**To Reproduce**
1. Navigate to Orders tab.
2. Tap on search icon.
3. Start typing a search term.
4. In some instances, the keyboard will be dismissed before you finish typing the search term.

**Expected behavior**
The keyboard should not be dismissed in this instance, it should remain until the user dismisses it themselves with the downward arrow button or with the search key.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x ] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Nokia 7.2
- Android version: Android 10
- WooCommerce Android version: 6.0-rc-2

Also was able to recreate the issue on my test device:

- Device: Pixel 3
- Android Version: Android 11
- WooCommerce Android version: 6.0-rc-2

**WordPress Environment**
Will add SSR as a comment.

3758472-zd-woothemes

**Comment by eduardozulian** [comment]
Adding the SSR:

```
### WordPress Environment ###

WordPress address (URL): https://www.boholifestylestore.nl
Site address (URL): https://www.boholifestylestore.nl
WC Version: 4.9.2
REST API Version: ✔ 4.9.2
WC Blocks Version: ✔ 4.0.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.8.3
Log Directory Writable: ✔
WP Version: 5.6.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: nl_NL
External object cache: –

### Server Environment ###

Server Info: nginx/1.15.8
PHP Version: 7.4.14
PHP Post Max Size: 64 MB
PHP Time Limit: 120
PHP Max Input Vars: 3000
cURL Version: 7.74.0
NSS/3.53.1

SUHOSIN Installed: –
MySQL Version: 5.6.51
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.9.2
WC Database Prefix: wp_
Totale databasegrootte: 399.15MB
Database datagrootte: 219.23MB
Database index grootte: 179.92MB
wp_woocommerce_sessions: Data: 3.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 1.52MB + Index: 0.33MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 7.52MB + Index: 6.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_404_to_301: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_actionscheduler_actions: Data: 4.45MB + Index: 3.30MB + Engine InnoDB
wp_actionscheduler_claims: Data: 1.52MB + Index: 1.52MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 2.52MB + Index: 3.03MB + Engine InnoDB
wp_ac_abandoned_cart_history_lite: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_ac_email_templates_lite: Data: 0.03MB + Index: 0.00MB + Engine InnoDB
wp_ac_guest_abandoned_cart_history_lite: Data: 0.30MB + Index: 0.00MB + Engine InnoDB
wp_ac_sent_history_lite: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_adtribes_my_conversions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_atum_log_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_berocket_termmeta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_braapf_product_stock_status_parent: Data: 0.14MB + Index: 0.08MB + Engine InnoDB
wp_braapf_product_variation_attributes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_braapf_term_taxonomy_hierarchical: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_braapf_variation_attributes: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 6.52MB + Index: 8.08MB + Engine InnoDB
wp_ewwwio_images: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ewwwio_queue: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mollie_pending_payment: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_newsletter: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_newsletter_emails: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_newsletter_sent: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_newsletter_stats: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_newsletter_user_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 9.50MB + Index: 4.20MB + Engine InnoDB
wp_pmxi_files: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_hash: Data: 0.45MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_history: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_images: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_imports: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_posts: Data: 0.13MB + Index: 0.00MB + Engine InnoDB
wp_pmxi_templates: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_postmeta: Data: 127.67MB + Index: 113.11MB + Engine InnoDB
wp_posts: Data: 16.52MB + Index: 20.61MB + Engine InnoDB
wp_scalability_pro_cache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_scalability_pro_profiling: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_smush_dir_images: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.28MB + Index: 0.28MB + Engine InnoDB
wp_terms: Data: 0.31MB + Index: 0.48MB + Engine InnoDB
wp_term_relationships: Data: 3.52MB + Index: 4.53MB + Engine InnoDB
wp_term_taxonomy: Data: 0.38MB + Index: 0.91MB + Engine InnoDB
wp_usermeta: Data: 7.52MB + Index: 3.03MB + Engine InnoDB
wp_users: Data: 0.11MB + Index: 0.08MB + Engine InnoDB
wp_vtprd_purchase_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_vtprd_purchase_log_product: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_vtprd_purchase_log_product_rule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wcj_track_users: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcpdf_invoice_number: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.33MB + Index: 0.22MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 1.52MB + Index: 0.73MB + Engine InnoDB
wp_wc_order_stats: Data: 0.42MB + Index: 0.33MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.27MB + Index: 0.23MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.45MB + Index: 0.59MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wdp_orders: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wdp_order_items: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wdp_rules: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_webpages_data: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_webwinkelkeur_invite_error: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yith_wcwl: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yith_wcwl_lists: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_indexable: Data: 14.55MB + Index: 6.56MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.14MB + Index: 0.20MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.23MB + Index: 0.23MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_meta: Data: 1.52MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

acf-field: 1
acf-field-group: 1
amn_exact-metrics: 2
attachment: 9507
blocks: 12
br_labels: 1
br_product_filter: 10
custom_css: 2
gfw_report: 1
jetpack_migration: 2
nav_menu_item: 97
o-discount: 1
o-list: 1
oembed_cache: 148
page: 39
post: 111
product: 3020
revision: 259
shop_coupon: 13
shop_order: 3157
shop_order_refund: 473
shortcoder: 3
vc_grid_item: 1
wccaf: 2
wcccf: 15214
wpcf7_contact_form: 1
wphb_minify_group: 4
yith_wcps_type: 5

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (43) ###

404 to 301 - Redirect, Log and Notify 404 Errors: door Joel James – 3.0.7
Admin Filter BY Custom Fields: door Bainternet – 1.0
Amount Left for Free Shipping for WooCommerce: door WPFactory – 2.0.0
Antispam Bee: door pluginkollektiv – 2.9.3
Contact Form 7: door Takayuki Miyoshi – 5.3.2
WooCommerce Continue Shopping: door HappyKite – 1.4.3
Disable Cart Fragments: door Optimocha – 1.21
HSTS Ready: door Manu225 – 1.02
HTTP Headers: door Dimitar Ivanov – 1.18.1
Jetpack door WordPress.com: door Automattic – 9.4
Judge.me Product Reviews for WooCommerce: door Judge.me – 1.3.8
Loco Translate: door Tim Whitlock – 2.5.0
MainWP Child: door MainWP – 4.1.4
Mollie-betalingen voor WooCommerce: door Mollie – 6.1.0
OnSale Page for WooCommerce: door wpgenie – 1.0.11
SendCloud | Smart Shipping Service: door SendCloud B.V. – 1.1.3
Tawk.to Live Chat: door Tawkto – 0.4.3
Use Any Font: door Dnesscarkey – 6.0.4
Warpdrive: door Savvii  – 2.10.4
WC Duplicate Order: door Jamie Gill – 1.7
WooCommerce NL Postcode Checker: door Ewout Fernhout – 2.5.6
WebwinkelKeur: door Albert Peschar – 3.4
Woo Admin Filter By Stock: door EXTREME IDEA LLC – 1.0.6
Woo Out Of Stock Products: door Andrew Gunn – 2.3
Product Feed PRO for WooCommerce: door AdTribes.io – 9.6.1
Verlaten Winkelwagen Lite voor WooCommerce: door Tyche software – 5.8.6
Advanced AJAX Product Filters for WooCommerce: door BeRocket – 1.5.2.9
WooCommerce Free Gift: door themegrids – 4.0.17
WooCommerce Google Ads Conversion Tracking: door Wolf+Bär Agency – 1.8.15
WooCommerce Google Analytics Integration: door WooCommerce – 1.5.1
WooCommerce Google Ads Dynamic Remarketing: door Agentschap Wolf + Bär – 1.7.18
More Sorting Options for WooCommerce: door WP Wham – 3.2.7
WooCommerce PDF Invoices & Packing Slips: door Ewout Fernhout – 2.8.2
WooCommerce: door Automattic – 4.9.2 (update naar versie 5.0.0 is beschikbaar)
Woosa - bol.com voor WooCommerce: door Woosa – 1.4.2
Yoast SEO: door Team Yoast – 15.8
WP All Import Pro: door Soflyy – 4.6.5
WP-Mail-SMTP: door WPForms – 2.6.0
WP Rocket | Exclude Files from Async CSS: door WP Rocket Support Team –
WP Rocket: door WP Media – 3.8.5
Smush: door WPMU DEV – 3.8.3
WP All Import - WooCommerce Add-On Pro: door Soflyy – 3.2.5
YITH WooCommerce Product Slider Carousel Premium: door YITH – 1.0.42

### Inactive Plugins (2) ###

Facebook for WooCommerce: door Facebook – 2.3.0
YITH WooCommerce Product Bundles: door YITH – 1.2.4

### Dropin Plugins (1) ###

advanced-cache.php: advanced-cache.php

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)
yith_bundle (yith_bundle)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Winkelbasis: #364 - /shop/
Winkelmand: #365 - /cart/
Afrekenen: #366 - /checkout/
Mijn account: #8 - /my-account/
Algemene voorwaarden: ❌ Pagina niet ingesteld

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.13.1
Parent Theme Author URL: http://www.uxthemes.com/
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/terms.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product-cat.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome-child/woocommerce/emails/customer-completed-order.php
flatsome-child/woocommerce/emails/plain/customer-completed-order.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php
flatsome/woocommerce/content-product_cat.php


### More Sorting Settings ###

More Sorting for WooCommerce: yes
Custom Sorting: yes
Sort by date (ascending): 
Sort by date (descending): 
Sort by title: A to Z: Sorteer op naam : A to Z
Sort by title: Z to A: Sorteer op naam: Z to A
Sort by slug: A to Z: 
Sort by slug: Z to A: 
Sort by SKU: low to high: 
Sort by SKU: high to low: 
Sort by SKU: no
Sort by stock quantity: low to high: Sorteer op voorraad: laag naar hoog
Sort by stock quantity: high to low: Sorteer op voorraad: hoog naar laag
Sort by total sales: low to high: 
Sort by total sales: high to low: 
Sort by last modified date: oldest to newest: 
Sort by last modified date: newest to oldest: 
Sort by author ID (ascending): 
Sort by author ID (descending): 
Sort by product ID (ascending): 
Sort by product ID (descending): 
Sort by number of comments (ascending): 
Sort by number of comments (descending): 
Random sorting: 
No sorting: 
Custom Meta Sorting: no
Total Options: 1
Enable/Disable: yes
Type: meta_value_num
Meta Key: _stock_quantity
Secondary Sorting: none
Order: asc
Parameter: >0
Title: Op Voorraad
Default Sorting Options: no
Standaard sortering: Standaard sortering
Sorteer op populariteit: Sorteer op populariteit
Remove "Sorteer op populariteit": no
Op gemiddelde waardering sorteren: Op gemiddelde waardering sorteren
Remove "Op gemiddelde waardering sorteren": no
Sort by newness: Sort by newness
Remove "Sort by newness": no
Sorteer op prijs: laag naar hoog: Sorteer op prijs: laag naar hoog
Remove "Sorteer op prijs: laag naar hoog": no
Sorteer op prijs: hoog naar laag: Sorteer op prijs: hoog naar laag
Remove "Sorteer op prijs: hoog naar laag": no
Rearrange Sorting: no
Rearrange Sorting: menu_order
popularity
rating
date
price
price-desc
date-asc
date-desc
title-asc
title-desc
name-asc
name-desc
sku-asc
sku-desc
stock_quantity-asc
stock_quantity-desc
total_sales-asc
total_sales-desc
modified-asc
modified-desc
author-asc
author-desc
product_id-asc
product_id-desc
comment_count-asc
comment_count-desc
rand
none
Remove All Sorting (Including WooCommerce Default): no
Enable/Disable: no
Theme: avada

### Action Scheduler ###

Afgerond: 10.934
Oldest: 2021-01-23 08:25:26 +0000
Newest: 2021-02-22 15:20:31 +0000

Mislukt: 234
Oldest: 2020-11-09 17:49:10 +0000
Newest: 2020-11-09 18:57:11 +0000

In afwachting: 4
Oldest: 2021-02-22 15:35:31 +0000
Newest: 2021-02-28 10:31:16 +0000
```

**Comment by mdrockwell** [comment]
The same user, in the same ticket had another issue that might be related — both involve the keyboard being dismissed from the order search interface unexpectedly.

Steps to reproduce:
1. From Orders tab, tap to open an order.
2. Tap back button to return to Orders view.
3. Tap on search icon.
4. After a moment, the keyboard will dismiss.

Initial report from the same device setup:

Device: Nokia 7.2
Android version: Android 10
WooCommerce Android version: 6.0-rc-2

And experienced it on my test device too:

Device: Pixel 3
Android Version: Android 11
WooCommerce Android version: 6.0-rc-2

**Comment by nbradbury** [comment]
Looks like this is a side effect of the keyboard being dismissed when the query has been submitted after a delay in typing. It's easy to reproduce this:

- Tap the order search icon
- Start typing a search phrase
- Pause for a second
- Note that the search is submitted and keyboard is hidden

Problematic line is [here](https://github.com/woocommerce/woocommerce-android/blob/5e8ca2f636f92260d29c5467256df1c4a3def812/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/list/OrderListFragment.kt#L384). Note that the product list search does _not_ have this issue.
